doozer/doozerlib/backend/konflux_client.py

```
@@ -50,6 +55,111 @@ def get_common_runtime_watcher_labels() -> Dict[str, str]:
+    with _TEMPLATE_REPO_LOCK:
+        if cache_key in _TEMPLATE_REPO_CACHE:
+            repo_path = _TEMPLATE_REPO_CACHE[cache_key]
+            if repo_path.exists():
+                LOGGER.debug(f"Using cached template repo at {repo_path}")
+                # Fetch latest and checkout the ref
+                try:
+                    exectools.cmd_assert(f'git -C {repo_path} fetch --all --prune', retries=3)
+                    exectools.cmd_assert(f'git -C {repo_path} checkout {ref}', retries=3)
+                    # If ref is a branch, pull latest
+                    exectools.cmd_gather(f'git -C {repo_path} pull --ff-only 2>/dev/null || true')
+                    return repo_path
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)